elm/src/main/scala/io/scalac/elm/state/ElmWallet.scala

```diff
@@ -81,16 +83,29 @@ case class ElmWallet(secret: PrivateKey25519 = generateSecret(),
   override def historyTransactions: Seq[WalletTransaction[PublicKey25519Proposition, ElmTransaction]] = ???
 
   override def scanPersistent(modifier: ElmBlock): ElmWallet = {
-    // increase balance by confirmed outputs
     val transactions = modifier.transactions.getOrElse(Nil)
 
-    transactions.foldLeft(this) { (w, tx) =>
-      val outs = tx.outputs.filter(_.proposition.pubKeyBytes.key == pubKeyBytes)
-      val increasedOutputs = outs.foldLeft(chainTxOutputs)((m, o) => m + (o.id.key -> TimedTxOutput(o, tx.timestamp)))
-      val sum = outs.map(_.value).sum
-
+    // increase balance by confirmed outputs
+    val increasedWallet = {
```

Comment:
Transactions from other nodes may only increase _this wallet's_ funds. Do you mean transactions to other nodes? If so, that's already taken care of in `scanOffchain`. It's somewhat unintuitive, and also it means you cannot reconstruct a wallet simply by applying a blockchain to an empty wallet. So we may need to rethink this at some point, but for now it seems to work.